libs/src/utils/multiProvider.js

```diff
@@ -0,0 +1,44 @@
+const { promisify, callbackify } = require('util')
+const Web3 = require('../web3')
+const { shuffle } = require('lodash')
+
+class MultiProvider extends Web3.providers.HttpProvider {
+  // web3 consumes a provider object on initialization
+  // ref: https://github.com/ChainSafe/web3.js/blob/1.x/packages/web3/types/index.d.ts#L31
+  // which references: https://github.com/ChainSafe/web3.js/blob/1.x/packages/web3-core/types/index.d.ts#L436
+  // MultiProivder implements AbstractProvider which can be consumed by web3
```

Comment:
this looks great. it doesn't implement AbstractProvider though, can you fix

also see JSDoc style here https://jsdoc.app/howto-es2015-classes.html